Assets/Scripts/PlayerController.cs

```diff
@@ -45,24 +53,32 @@ public void OnDisable()
             input.Disable();
         }
 
+        public void Start()
+        {
+            // ===== temporary until we decide on a specific gravity value =====
+            originalGravityScale = rigidBody.gravityScale;
+            // =================================================================
+        }
+
         public void FixedUpdate()
         {
+            // ===== temporary until we decide on a specific gravity value =====
+            rigidBody.gravityScale = localGravity * originalGravityScale;
+            // =================================================================
+
+            // While there is no explicit speed cap, horizontal drag will create an artificial one.
+            rigidBody.velocity = new Vector2(rigidBody.velocity.x * (1.0f - horizontalDrag), rigidBody.velocity.y);
+
             // Horizontal movement
             rigidBody.AddForce(moveDirection.x * Vector2.right * speed);
```

Comment:
Unity's own physics system internally scales with time, so that is not a concern.

Also, implementing a drag factor instead of a hard speed limit makes it handle fast speeds much more naturally. There is no distinguishing between a drag factor and kinematic movement at low speeds. The current version doesn't just accelerate to ridiculous speeds and fly off the map. You can test it yourself in editor.

Using unity's built-in AddForce() instead of directly controlling values in Rigibody2D is much less prone to bugs in movement as well, since the less logic we are handling, the easier it is to track down what is wrong.

I'd just let unity handle this instead of trying to homebrew a physics solution.